library/Vanilla/DateFilterSphinxSchema.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla;
+
+/**
+ * Validate date period for Sphinx usage
+ */
+class DateFilterSphinxSchema extends DateFilterSchema {
+    /**
+     * If the parameter value is a valid date filter value,
+     * return a structure: startDate, endDate, exclude flag.
+     * Useful for Sphinx setFilterRange
+     *
+     * @param mixed $dateData The decoded and validated date data.
+     *
+     * @throws Exception If dateData structure is wrong throw exception.
+     *
+     * @return array Structure: DateTime startDate, DateTime endDate, Boolean inclusive
+     */
+    public static function dateFilterRange(array $dateData): array {
+        $validOperators = ['=', '>', '<', '>=', '<=', '[]', '()'];
+        $result = [
+            'startDate' => null,
+            'endDate' => null,
```

Comment:
When I'd asked about nulling these values out, when they were in `DateFilterSchema`, you said: 

> Because Sphinx setFilterRange method accepts only integers [...] so we need to send min and max value anyway.

Is that not the case? My only real objection to this method before was that it had Sphinx-specific stuff in it, but it looks like you've removed that by nulling these fields. If you still want to keep this method in a separate class, that's cool too, but I want to make sure these are valid values.

Reply:
`null`s are not valid values, but I handle them on Knowledge side